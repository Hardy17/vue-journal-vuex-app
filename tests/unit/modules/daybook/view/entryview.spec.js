import { shallowMount } from "@vue/test-utils"
import { createStore } from "vuex"

import EntryView from "@/modules/daybook/views/EntryView"
import Swal from "sweetalert2"
import journalModule from "@/modules/daybook/store/journal"
import { journalState } from "../mocks/test-journal-state"


const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journalModule: {
        ...journalModule,
        state: { ...initialState }
      }
    }
  })

jest.mock("sweetalert2", () => ({
  fire: jest.fn(),
  showLoading: jest.fn(),
  close: jest.fn()
}))
describe("pruebas sobre el entryview", () => {
  const store = createVuexStore(journalState)
  store.dispatch=jest.fn()
  const mockRouter = {
    push: jest.fn()
  }
  let wrapper
  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallowMount(EntryView, {
      props: {
        id: journalState.entries[0].id
      },
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [store]
      }
    })
  })
  test("debe hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test("debe sacar al usuario porque el id no existe", () => {
    const wrapper = shallowMount(EntryView, {
      props: {
        id: "Este ID no existe"
      },
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [store]
      }
    })
    
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" })
  })
  test("debe mostrar la entrada correctamente", () => {
    expect(mockRouter.push).not.toHaveBeenCalled()
  })

  test("debe de borrar la entrada  y salir", (done) => {
    Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }))

    wrapper.find(".btn-danger").trigger("click");

    expect(Swal.fire).toHaveBeenCalledWith({
      title: "Â¿Estas seguro?",
      text: "Una vez borrado, no se puede recuperar!!",
      showDenyButton: true,
      confirmButtonText: "Si, estoy seguro"
    })

    setTimeout(() => {
        expect(store.dispatch).toHaveBeenCalledWith('journalModule/deleteEntry', '-MwWv5wWZYBwjzGyFUZ0')
        expect(mockRouter.push).toHaveBeenCalled()
        done()
    }, 1)
  })
})
