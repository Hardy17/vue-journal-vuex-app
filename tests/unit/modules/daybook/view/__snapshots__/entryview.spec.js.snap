// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre el entryview debe hacer match con el snapshot 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">22</span><span class="mx-1 fs-3">Febrero</span><span class="mx-2 fs-4 fw-light">2022, Martes</span></div>
  <div><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="Â¿Que sucedio hoy?"></textarea></div>
<!--v-if-->
<!--v-if-->
<fab-stub icon="fa fa-save"></fab-stub>
`;
